src/main/java/seedu/address/logic/parser/EntryBookArchivesParser.java

```diff
@@ -0,0 +1,45 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.regex.Matcher;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.HelpCommand;
+import seedu.address.logic.commands.UnarchiveCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Represents the parser for the archive-context.
+ * It successfully parses a command if and only if the command is archive-context command.
+ */
+public class EntryBookArchivesParser extends EntryBookParser {
+
+    /**
+     * Parses user input into command for execution.
+     * Parses successfully if and only if the command is archive-context command.
+     *
+     * @param userInput full user input string
+     * @return the command based on the user input
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public Command parseCommand(String userInput) throws ParseException {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
+        if (!matcher.matches()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String commandWord = matcher.group("commandWord");
+        final String arguments = matcher.group("arguments");
```

Comment:
These few lines are duplicated. Since most (all?) commands follow the same `commandWord arguments` format maybe we extract everything below into an abstract method that only takes in `commandWord` and `arguments`?

Reply:
I don't think we need to do this because we should give parsers the freedom to parse things however they want.

For example we might want to have the ability to directly input the link into the command line to add it, and in this case there would be no `commandWord`.

Reply:
Makes sense... Sticking with this design is more robust.